network/p2p/config.go

```diff
@@ -14,7 +14,6 @@ type Config struct {
 	HostAddress         string
 	HostDNS             string
 	HostID              peer.ID
-	TopicName           string
-	Topic               *pubsub.Topic
-	Sub                 *pubsub.Subscription
+	Topics map[string]*pubsub.Topic
```

Comment:
any reason we need to hold topics and pubsubs in the struct? isn't just holding pubsub is enough?

Reply:
we need the topics for the broadcasting, pubsub is the gossip